contracts/ERC721License.sol

```diff
@@ -1,38 +1,85 @@
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.6.0;
+pragma solidity ^0.8.0;
 
-import "./ERC721Upgradeable.sol";
-import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/security/Pausable.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
 
-contract ERC721License is ERC721Upgradeable, AccessControlUpgradeable {
-    bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
+contract ERC721License is ERC721, Pausable, AccessControl {
+    bytes32 public constant PAUSE_ROLE = keccak256("PAUSE_ROLE");
+    bytes32 public constant MINT_ROLE = keccak256("MINT_ROLE");
+    bytes32 public constant BURN_ROLE = keccak256("BURN_ROLE");
+    bytes32 public constant OPERATOR_ROLE = keccak256("OPERATOR_ROLE");
 
-    function initialize(address admin) public {
-        __ERC721_init("GeoWebLicense", "GEO");
+    constructor() ERC721("Geo Web Parcel License", "GEOL") {
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        _setupRole(PAUSE_ROLE, msg.sender);
+    }
 
-        _setupRole(ADMIN_ROLE, admin);
+    /**
+     * @notice Pause the contract. Pauses transfers.
+     * @custom:requires PAUSE_ROLE
+     */
+    function pause() external onlyRole(PAUSE_ROLE) {
+        _pause();
     }
 
-    function mintLicense(
-        address to,
-        uint256 tokenId
-    ) external {
-        require(hasRole(ADMIN_ROLE, msg.sender), "Caller is not admin");
+    /**
+     * @notice Unpause the contract.
+     * @custom:requires PAUSE_ROLE
+     */
+    function unpause() external onlyRole(PAUSE_ROLE) {
+        _unpause();
+    }
 
+    /**
+     * @dev MINT_ROLE can mint a new license
+     */
+    function safeMint(address to, uint256 tokenId)
+        external
+        onlyRole(MINT_ROLE)
+    {
         _safeMint(to, tokenId);
     }
 
-    function _isApprovedOrOwner(address spender, uint256 tokenId)
-        internal
+    /**
+     * @dev BURN_ROLE can burn a license
+     */
+    function burn(uint256 tokenId) external onlyRole(BURN_ROLE) {
+        _burn(tokenId);
+    }
```

Comment:
does burning a license mean that the parcel would be back to being `SimpleETHClaimer#claim(able)` or would the next user still `ETHPurchaser#purchase` the parcel?

Reply:
This specific method would not affect the "availability" of the land, but just burns the NFT.

[This method](https://github.com/Geo-Web-Project/core-contracts/blob/main/contracts/GeoWebParcel.sol#L90) will make the parcel available again to be claimed on `SimpleETHClaimer#claim`